diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -89,42 +95,82 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     private lateinit var positionByOffset: (Int) -> Pair<Int, Int>
 
     override fun logic(node: ASTNode) {
-        if (node.elementType == KtFileElementType.INSTANCE) {
-            node.getChildren(null).forEach {
-                if (it.elementType != PACKAGE_DIRECTIVE && it.elementType != IMPORT_LIST) {
-                    checkLength(it, configuration)
+        var isFixedSmthInPreviousStep: Boolean
+
+        do {
+            isFixedSmthInPreviousStep = false
+
+            if (node.elementType == KtFileElementType.INSTANCE) {
+                node.getChildren(null).forEach {
+                    if (it.elementType != PACKAGE_DIRECTIVE && it.elementType != IMPORT_LIST) {
+                        val isFixedSmthInChildNode = checkLength(it, configuration)
+
+                        if (!isFixedSmthInPreviousStep && isFixedSmthInChildNode) {
+                            isFixedSmthInPreviousStep = true
+                        }
+                    }
                 }
             }
-        }
+        } while (isFixedSmthInPreviousStep)
```

Comment:
maybe add some limit on fixing to avoid infinite loop?

Reply:
btw, long time ago, when I refactor this rule, we discuss with @akuleshov7 , about loop in this rule
I dont see anything bad in this, but Andrey said not to do it for some reasons, that I forget

may be he will remember the reason